presto-main/src/test/java/io/prestosql/tests/LogTestDurationListener.java

```diff
@@ -117,10 +121,25 @@ else if (finished.get()) {
 
     private static void dumpAllThreads(String message)
     {
-        LOG.warn("%s\n\nFull Thread Dump:\n%s", message,
-                Arrays.stream(getThreadMXBean().dumpAllThreads(true, true))
-                        .map(ThreadInfo::toString)
-                        .collect(joining("\n")));
+        ListMultimap<String, ThreadInfo> threadInfos = Multimaps.index(
+                Arrays.asList(getThreadMXBean().dumpAllThreads(true, true)),
+                threadInfo -> Arrays.toString(threadInfo.getStackTrace()));
```

Comment:
We used `ThreadInfo::toString`, now we print only `getStackTrace`. This won't include locks information.

Reply:
Yes. This is the cost. Do you find this useful?